dbt/adapters/glue/impl.py

```diff
@@ -508,6 +508,50 @@ def create_csv_table(self, model, agate_table):
         except Exception as e:
             logger.error(e)
 
+    def _update_additional_location(self, target_relation, location):
+        session, client = self.get_connection()
+        table_input = {}
+        try:
+            table_input = client.get_table(
+                DatabaseName=f'{target_relation.schema}',
+                Name=f'{session.credentials.delta_athena_prefix}_{target_relation.name}',
+            ).get("Table", {})
+        except client.exceptions.EntityNotFoundException as e:
+            logger.debug(e)
+            pass
+        except Exception as e:
+            logger.error(e)
+
+        try:
+            # removing redundant keys from table_input
+            del table_input['DatabaseName'], \
+                table_input['CreateTime'], \
+                table_input['UpdateTime'], \
+                table_input['CreatedBy'], \
+                table_input['IsRegisteredWithLakeFormation'], \
+                table_input['CatalogId'], \
+                table_input['VersionId']
+
+            if 'AdditionalLocations' not in table_input['StorageDescriptor']:
+                table_input['StorageDescriptor']['AdditionalLocations'] = [location]
+            else:
+                if location not in table_input['StorageDescriptor']['AdditionalLocations']:
+                    table_input['StorageDescriptor']['AdditionalLocations'] += location
```

Comment:
What is `location` type? If it's a string it's end up iterating through letters:
```python
arr = ['loc1']
location = 'loc2'
arr += location
arr
Out[26]: ['loc1', 'l', 'o', 'c', '2']
```
I'd suggest to use `append` or `extend` depending on `location` type.